src/components/sheet/sheet-action-buttons/BuyActionButton.js

```diff
@@ -3,17 +3,17 @@ import React, { useCallback } from 'react';
 import SheetActionButton from './SheetActionButton';
 import { analyticsV2 } from '@/analytics';
 import showWalletErrorAlert from '@/helpers/support';
-import { useAccountSettings, useExpandedStateNavigation, useWallets } from '@/hooks';
+import { useWallets } from '@/hooks';
 
 import Routes from '@/navigation/routesNames';
 import { useRoute } from '@react-navigation/native';
+import { useNavigation } from '@/navigation';
 
 function BuyActionButton({ color: givenColor, asset, ...props }) {
   const { colors } = useTheme();
   const color = givenColor || colors.paleBlue;
-  const navigate = useExpandedStateNavigation(null, true, asset);
```

Comment:
we had logic in the `useExpandedStateNavigation` that would also check if the wallet was `readOnly` and throw an alert if so (same for all the other buttons that used to use the `useExpandedStateNavigation` hook